internal/planner/scan.go

```diff
@@ -325,35 +325,53 @@ func (p *Planner) Scan(
 type multiScanNode struct {
 	scanNode   *scanNode
 	numReaders int
-	numCalls   int
-
-	lastBool bool
-	lastErr  error
+	nextCount  int
+	initCount  int
+	startCount int
+	closeCount int
+
+	nextResult bool
+	nextErr    error
+	initErr    error
```

Comment:
question: Why are you storing these as state?  And why are they stored separately (`nextErr`, `initErr`, etc instead of just `err`)?

Reply:
left only 1 err